src/olgm/listener/PropertyListener.js

```diff
@@ -0,0 +1,39 @@
+import AbstractListener from './AbstractListener';
+import {unByKey} from 'ol/Observable';
+
+class PropertyListener extends AbstractListener {
+  /**
+   * Listener for OL object properties. Has utilities for listening on the property value.
+   * @param {module:ol/Object~BaseObject} target Object with a property
+   * @param {?module:ol/Object~BaseObject} oldTarget The previous object with a property
+   * @param {string} key Property key
+   * @param {function} listen Takes the current and previous property value as arguments.
+   * Called on initialization and when the property value changes.
+   * On initialization, the previous property value will be taken from the old target, if any.
+   * Can return an AbstractListener which will be unlistened when the property value changes.
+   */
+  constructor(target, oldTarget, key, listen) {
+    super();
+
+    this.targetListenerKey = target.on('change:' + key, e => {
+      if (this.innerListener) {
+        this.innerListener.unlisten();
+      }
+      this.innerListener = listen(e.target.get(e.key), e.oldValue);
+    });
+
+    this.innerListener = listen(target.get(key), oldTarget && oldTarget.get(key));
```

Comment:
type def

PropertyListener inherits AbstractListener.  Suggestion: it could inherit Listener instead, and instead of defining `this.innerListener`, would it make sense to re-use `this.listenerKey` ?  Then, `unlisten` of PropertyListener could first call its parent, then unlisten `this.targetListenerKey`.